### PR TITLE
bear 3.0.21

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **bear/3.0.21**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/bear/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+required_conan_version = ">=1.53.0"
+
+
+class BearConan(ConanFile):
+    name = "bear"
+    description = "Bear is a tool that generates a compilation database for clang tooling"
+    license = "GPL-3.0-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rizsotto/Bear"
+    topics = ("compilation", "database")
```
Comment (jwillikers) [inline recipes/bear/all/conanfile.py:21]:
```suggestion
    topics = ("clang", "compilation", "database", "llvm")
```

File: recipes/bear/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built on windows.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        # BUILD_SHARED_LIBS and POSITION_INDEPENDENT_CODE are automatically parsed when self.options.shared or self.options.fPIC exist
```
Comment (jwillikers) [inline recipes/bear/all/conanfile.py:70]: I think you can remove these comments.

File: recipes/bear/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from conan import ConanFile
+
+
+# It will become the standard on Conan 2.x
```
Comment (jwillikers) [inline recipes/bear/all/test_package/conanfile.py:4]: I think this comment can be removed.

**Review by jwillikers — CHANGES_REQUESTED** [review]
I think you should remove some of the comments, but otherwise looks good.

File: recipes/bear/all/conanfile.py (hunk 2 above)
Comment (ericLemanissier) [inline recipes/bear/all/conanfile.py:70]:
```suggestion
```

File: recipes/bear/all/test_package/conanfile.py (hunk 3 above)
Comment (ericLemanissier) [inline recipes/bear/all/test_package/conanfile.py:4]:
```suggestion
```

**Review by jwillikers — APPROVED** [review]

File: recipes/bear/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+sources:
+  "3.0.21":
+    url: "https://github.com/rizsotto/Bear/archive/refs/tags/3.0.21.tar.gz"
+    sha256: "0c949a6a907bc61a1284661f8d9dab1788a62770c265f6142602669b6e5c389d"
+patches:
+  "3.0.21":
+    - patch_file: "patches/7de12540948cd5e0ee5f966b5cc383bc0c0e5fe9.diff"
```
Comment (prince-chrismc) [inline recipes/bear/all/conandata.yml:7]: I love patch fields 😍 hint hint

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
There is something wrong in this recipe. It let upstream CML downloads stuff at build time like fmt:

```
-- Looking for fmt dependency -- not found
-- Looking for spdlog dependency
-- Conan: Component target declared 'spdlog::spdlog'
-- Conan: Component target declared 'fmt::fmt'
-- Looking for spdlog dependency -- found
-- Looking for gRPC::grpc++ dependency
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for modules 'protobuf>=3.11;grpc++>=1.26'
--   Found protobuf, version 3.21.4
--   Found grpc++, version 1.50.1
-- Looking for gRPC::grpc++ dependency -- found
-- Looking for protoc
-- Looking for protoc -- found
-- Looking for grpc_cpp_plugin
-- Looking for grpc_cpp_plugin -- found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release
bear/3.0.21: CMake command: cmake --build "/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release" '--' '-j3'

----Running------
> . "/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/generators/conanbuild.sh" && cmake --build "/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release" '--' '-j3'
-----------------
Scanning dependencies of target grpc_dependency
Scanning dependencies of target googletest_dependency
Scanning dependencies of target nlohmann_json_dependency
[  0%] Built target grpc_dependency
[  0%] Built target googletest_dependency
[  0%] Built target nlohmann_json_dependency
Scanning dependencies of target fmt_dependency
Scanning dependencies of target spdlog_dependency
[  5%] Creating directories for 'fmt_dependency'
[  5%] Built target spdlog_dependency
[ 11%] Performing download step (download, verify and extract) for 'fmt_dependency'
-- Downloading...
   dst='/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release/subprojects/Download/fmt_dependency/9.1.0.tar.gz'
   timeout='none'
-- Using src='https://github.com/fmtlib/fmt/archive/9.1.0.tar.gz'
-- verifying file...
       file='/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release/subprojects/Download/fmt_dependency/9.1.0.tar.gz'
-- Downloading... done
-- extracting...
     src='/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release/subprojects/Download/fmt_dependency/9.1.0.tar.gz'
     dst='/home/conan/w/prod/BuildSingleReference/.conan/data/bear/3.0.21/_/_/build/394f545e9cc458ec75d78c70083d1b2301255007/build/Release/subprojects/Source/fmt_dependency'
```

**Pull request merged**